File: Programmers/이상한 문자 만들기.js
// https://programmers.co.kr/learn/courses/30/lessons/12930

function solution(s) {
  return s
    .split(" ")
    .map((n) => change(n))
    .join(" ");
}

const change = (s) => {
  let answer = "";
  for (let i = 0; i < s.length; i++) {
    const target = s.charCodeAt(i);
    if (i % 2 && target >= 65 && target <= 90) {
      answer += String.fromCharCode(target + 32);
      continue;
    }
    if (!(i % 2) && target >= 97 && target <= 122) {
      answer += String.fromCharCode(target - 32);
      continue;
    }
    answer += s[i];
  }
  return answer;
};

const s = "try hello world";
// const s = "azAZ";
// const s = " ";

console.log(solution(s));

File: Programmers/이상한 문자 만들기.swift
// https://programmers.co.kr/learn/courses/30/lessons/12930?language=swift
// 이상한 문자 만들기

func solution(_ s:String) -> String {
    var splitted: [String] = s.components(separatedBy: " ")
    var answer: String = ""
    for (index, str) in splitted.enumerated() {
        for i in 0..<str.count {
            if i % 2 == 1 {
                answer += str[str.index(str.startIndex, offsetBy: i)].lowercased()
            } else {
                answer += str[str.index(str.startIndex, offsetBy: i)].uppercased()
            }
        }
        if index != splitted.count-1 { answer += " " }
    }
    return answer
}

let s: String	= "try hello world"	
// "TrY HeLlO WoRlD"

print(solution(s))